pydantic/dataclasses.py

```diff
@@ -162,7 +162,7 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         original_cls = cls
 
         config_wrapper = _config.ConfigWrapper(config)
-        decorators = _decorators.DecoratorInfos.build(cls)
+        cls.__pydantic_decorators__ = _decorators.DecoratorInfos.build(cls)  # type: ignore
```

Comment:
Note that this PR https://github.com/pydantic/pydantic/pull/5874 will move to re-computing the decorators whenever they aren't present on the exact class. Not sure if that will cause issues with this change.

Reply:
(I just merged that PR)

Reply:
I was just cleaning up here: we used `decorators`  just as a temp variable and then assigned to `cls. __pydantic_decorators__`.

Reply:
the line immediately before the old version of this was:
```python
        cls = dataclasses.dataclass(  # type: ignore[call-overload]
            cls,
            init=init,
            repr=repr,
            eq=eq,
            order=order,
            unsafe_hash=unsafe_hash,
            frozen=frozen,
            **kwargs,
        )
```
does this not change the value of `cls`? (Just modify it?) That's why I thought there was a change to the semantics not just cleanup.

Reply:
let me just revert this then